src/BasicWL/ApplicableHeaderTradeSettlement.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tiime\CrossIndustryInvoice\BasicWL;
+
+use Tiime\EN16931\DataType\CurrencyCode;
+use Tiime\EN16931\DataType\Identifier\BankAssignedCreditorIdentifier;
+
+/**
+ * BG-19.
+ */
+class ApplicableHeaderTradeSettlement
+{
+    /**
+     * BT-90.
+     */
+    private ?BankAssignedCreditorIdentifier $creditorReferenceId;
+
+    /**
+     * BT-83.
+     */
+    private ?string $paymentReference;
+
+    /**
+     * BT-6.
+     */
+    private ?CurrencyCode $taxCurrencyCode;
+
+    /**
+     * BT-5.
+     */
+    private CurrencyCode $invoiceCurrencyCode;
+
+    /**
+     * BG-10.
+     */
+    private ?PayeeTradeParty $payeeTradeParty;
+
+    /**
+     * BG-16.
+     */
+    private ?SpecifiedTradeSettlementPaymentMeans $specifiedTradeSettlementPaymentMeans;
+
+    /**
+     * BG-23.
+     *
+     * @var non-empty-array<int, ApplicableTradeTax>
+     */
+    private array $applicableTradeTaxes;
+
+    /**
+     * BG-14.
+     */
+    private ?BillingSpecifiedPeriod $billingSpecifiedPeriod;
+
+    /**
+     * BG-20.
+     *
+     * @var array<int, SpecifiedTradeAllowance>
+     */
+    private array $specifiedTradeAllowance;
```

Comment:
Can you add an "s" at the end of the variable : "$specifiedTradeAllowances" please ?